src/main/java/pageclasses/CartPage.java

```diff
@@ -0,0 +1,112 @@
+package pageclasses;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class CartPage extends BasePage {
+
+    @FindBy(className = "item-info")
+    private List<WebElement> productList;
+
+    @FindBy(css = ".price .cart-price .price")
+    private List<WebElement> productPrices;
+
+    @FindBy(css = ".input-text.qty")
+    private List<WebElement> productQuantities;
+
+    @FindBy(css = ".subtotal .cart-price .price")
+    private List<WebElement> productSubtotals;
+
+    @FindBy(css = ".sub .price")
+    private WebElement cartSubtotal;
+
+    @FindBy(css = ".grand .price")
+    private WebElement grandTotal;
+
+    private static final By DISCOUNT = By.cssSelector("td[data-th='Discount'] .price");
+    private static final By TAX = By.cssSelector(".totals-tax .price");
+
+    public CartPage() {
+        super();
+    }
+
+    public NavBar getNavBar() {
+        return new NavBar();
+    }
+
+    public int getProductQuantity() {
+        int initialQuantity = 0;
+        for (WebElement productQuantity : productQuantities) {
+            int quantity = Integer.parseInt(productQuantity.getAttribute("value"));
+            initialQuantity += quantity;
+        }
+        return initialQuantity;
+    }
+
+    public boolean isProductSubtotalValid() {
+        for (int i = 0; i < productList.size(); i++) {
+            WebElement productPrice = productPrices.get(i);
+            double price = Double.parseDouble(productPrice.getText().replaceAll("[,$]", ""));
```

Comment:
create extractor for the money values. Ex.:
```
public String extractMoneyValue(String text) {
        String value = "";
        Pattern pattern = Pattern.compile("\\d+\\.?\\d{0,2}");
        Matcher matcher = pattern.matcher(text);
        if (matcher.find()) {
            value = matcher.group();
        }
        return value;
    }
```

```
public double extractDouble(String text) {
        return Double.parseDouble(Actions.extractMoneyValue(text));
    }
```
and delete all` .replaceAll("[,$]" `...